src/components/Nav.tsx

```diff
@@ -92,19 +93,21 @@ const Nav = () => {
 							</div>
 						</div>
 						<div className="right_navigations w-full ipad:w-max flex items-center justify-end gap-6">
-							<Link to={'/carts'}>
-								<div className="cart relative">
-									<IoCart className="text-3xl mobile:text-xl ipad:text-3xl text-primary-lightblue" />
-									{numberOfItem > 0 ? (
-										<span
-											data-testid="number-of-item"
-											className="bg-neutral-darkRed text-sm mobile:text-base p-1 absolute -top-3 -right-3 rounded-full flex items-center justify-center text-neutral-white font-semibold mobile:w-6 w-5 h-5 mobile:h-6"
-										>
-											{numberOfItem}
-										</span>
-									) : null}
-								</div>
-							</Link>
+							{fetchInfo()?.role === 'BUYER' && (
+								<Link to={'/carts'}>
```

Comment:
let's not fetch info in UI part, we can fetch them while component is rendering.

like const userData = fetchInfo()

userData.role ...... the we perform the task

Reply:
ooh i understand, let me fix it